### PR TITLE
Geocode and state picklists

### PR DESCRIPTION
**Pull request by gigxz**
* Add pick lists GEOCODE and STATE.
* Geocode is limited to geocodes in `ENV['RELEVANT_COC_STATE']`.
* State list is always the full list, but the `ENV['RELEVANT_COC_STATE']` is will be selected by default if no value is present (using `initial_selected`).
* Update COC list to include the coc code in the name for better searchability.
* Includes all HUD Geocodes from FY 2022 https://www.hud.gov/sites/dfiles/CPD/documents/FY-2022-GeoCodes-Report.pdf

Screen Shot 2022-11-01 at 4 59 53 PM

Screen Shot 2022-11-01 at 5 00 06 PM

### CODE REVIEW

File: drivers/hmis/app/graphql/types/forms/pick_list_option.rb [hunk 1]
```diff
@@ -19,7 +19,26 @@ def self.options_for_type(pick_list_type, user:)
...
+          {
+            code: obj['abbreviation'],
+            label: "#{obj['abbreviation']} - #{obj['name']}",
+            initial_selected: obj['abbreviation'] == ENV['RELEVANT_COC_STATE'],
+          }
+        end
+
+      when 'GEOCODE'
+        geocodes = JSON.parse(File.read("drivers/hmis/lib/pick_list_data/geocodes/geocodes-#{ENV['RELEVANT_COC_STATE']}.json"))
+        geocodes.map do |obj|
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/forms/pick_list_option.rb]: Probably not ideal to do this every time, we only need to load it into memory once. @eanders do you have a suggestion on how to cache here, given that its a class method?

File: drivers/hmis/app/graphql/types/forms/pick_list_option.rb [hunk 2]
```diff
@@ -19,7 +19,26 @@ def self.options_for_type(pick_list_type, user:)
...
+        states.map do |obj|
+          {
+            code: obj['abbreviation'],
+            label: "#{obj['abbreviation']} - #{obj['name']}",
+            initial_selected: obj['abbreviation'] == ENV['RELEVANT_COC_STATE'],
+          }
+        end
+
+      when 'GEOCODE'
+        geocodes = JSON.parse(File.read("drivers/hmis/lib/pick_list_data/geocodes/geocodes-#{ENV['RELEVANT_COC_STATE']}.json"))
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/forms/pick_list_option.rb]: Not sure we need to reload this each day, but if the process is relatively light, then that feels like a reasonable cache time.
```suggestion
        geocodes = Rails.cache.fetch(['GEOCODES', ENV['RELEVANT_COC_STATE']], expires_in: 1.days) do   
          JSON.parse(File.read("drivers/hmis/lib/pick_list_data/geocodes/geocodes-#{ENV['RELEVANT_COC_STATE']}.json"))
        end
```

Comment (eanders) [reply drivers/hmis/app/graphql/types/forms/pick_list_option.rb]: I missed the map below, but that can go in the cache too.

**Review by eanders — APPROVED** [review]

File: drivers/hmis/app/graphql/types/forms/pick_list_option.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/forms/pick_list_option.rb]: Updated! thanks

**Pull request merged**